crates/bevy_render/src/primitives/mod.rs

```diff
@@ -72,14 +72,26 @@ impl Sphere {
     }
 }
 
-/// A plane defined by a normal and distance value along the normal
+/// A plane defined by a normalized normal and distance value along the normal
 /// Any point p is in the plane if n.p = d
-/// For planes defining half-spaces such as for frusta, if n.p > d then p is on the positive side of the plane.
+/// For planes defining half-spaces such as for frusta, if n.p > d then p is on the positive side (inside) of the plane.
 #[derive(Clone, Copy, Debug, Default)]
 pub struct Plane {
     pub normal_d: Vec4,
 }
 
+impl Plane {
+    /// Constructs a `Plane` from a 4D vector whose first 3 components
+    /// are the normal and whose last component is d.
+    /// Ensures that the normal is normalized and d is scaled accordingly
+    /// so it represents the signed distance from the origin.
+    fn new(normal_d: Vec4) -> Self {
+        Self {
+            normal_d: normal_d * normal_d.xyz().length_recip(),
+        }
+    }
```

Comment:
```suggestion
    }

    /// `Plane` unit normal
    #[inline]
    pub fn normal(&self) -> Vec3 {
        self.normal_d.xyz()
    }

    /// Signed distance from the origin along the unit normal such that n.p + d = 0 for point p in
    /// the `Plane`
    #[inline]
    pub fn d(&self) -> f32 {
        self.normal_d.w
    }

    /// `Plane` unit normal and signed distance from the origin such that n.p + d = 0 for point p
    /// in the `Plane`
    #[inline]
    pub fn normal_d(&self) -> Vec4 {
        self.normal_d
    }
```
And then use these in the `Frustum` intersects sphere/obb functions below.